### PR TITLE
Fix timepicker in Finnish locales

### PR DESCRIPTION
**Pull request by eyelidlessness**
Fixes enketo/enketo-express#453. Opening as draft for now until the following are complete (unless we determine not to do the remaining work):

- [x] Use the [`Intl.DateTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) API rather than regex to determine time structure.
- [x] Use a locale-specific date string to determine whether a time *string* has a meridian component. This is necessary because you can't `new Date('01:00 pm')`.
- [x] Detect locale changes and updates the locale state used for the above accordingly.
- [x] Detect locale change to update widget state.
- ~~[ ] Format widget display state with locale-specific punctuation rather than hard coding the separator as `:`~~

### CODE REVIEW

**Comment by lognaturel** [comment]
Nice! Do you see a big advantage to doing the last two points in this PR vs treating them as separate issues?

Both seem nice to have but not critical to me so it all depends on whether you think they’re quick or not.

**Comment by eyelidlessness** [comment]
It was pretty straightforward to handle the language change. I decided to hold off on the separator though because it occurs to me there's probably other issues with that as well. Anyway, I think this is now ready for review!

File: test/spec/format.spec.js
Comment (lognaturel) [inline test/spec/format.spec.js:7]: I'm surprised not to see any additions here! Is there a test that would have failed prior to the change?

**Review by lognaturel — COMMENTED** [review]
Looks good to me! I'm just having trouble identifying a test that would have failed previously. Worth doing a sanity check with Express before merging?

File: test/spec/types.spec.js
Comment (eyelidlessness) [reply test/spec/types.spec.js:12]: Nope, typo + copypasta 🙃

File: test/spec/format.spec.js
Comment (eyelidlessness) [reply test/spec/format.spec.js:7]: Good call, I meant to actually add the case from the issue! I was focused on getting the setup/te~~e~~ardown switched over and forgot.

**Review by lognaturel — APPROVED** [review]
Looks great. @eyelidlessness mentioned some risk around date/times coming in from new submissiosn vs. existing submissions vs. defaults and whether those are all localized at the same time. As far as I can tell, the interaction with all of those is unchanged from the original regex-based solution so it doesn't seem like there's a lot of risk there to me. Still, I've made a note for QA to do some additional verification once it's available in Express!

**Pull request merged**
